### DISCUSSION TITLE
Ability to add random to playlist_songs

### DISCUSSION THREAD

**Discussion by jeison1946** [ampache/ampache#3566] [Ideas]
I am trying to list a ramdon song with the playlist_songs method but it always brings me the first one, it is possible

**Comment by mitchray** [comment]
I would recommend using [advanced_search](https://ampache.org/api/api-advanced-search/) with the following properties

```
type: 'song'
random: 1
limit: 1
operator: 'and'
rule_1: 'playlist'
rule_1_operator: 4
rule_1_input: '<YOUR PLAYLIST ID>'
```

**Reply by jeison1946** [reply]
I tried but it returns many songs

**Comment by lachlan-00** [comment]
i can add that. the functions on playlist are already there

parameter in api6 will be random with an int limit

**Comment by lachlan-00** [comment]
set in develop for api6. i'll update website docs soon

```
action=playlist_songs&auth=auth&filter=2184&limit=1&random=1
```

**Reply by lachlan-00** [reply]
set in the api now. might backport that to api5 as well
https://ampache.org/api/api-json-methods#playlist_songs

**Reply by jeison1946** [reply]
I made that request but it doesn't take me randomly, it always returns the first song in the playlist

**Reply by lachlan-00** [reply]
i've put it in patch5 branch for the next release and in develop/patch6 for api 5 and 6.

what branch are you on and what version are you handshaking with?

commit e181a13c4 and e369f501c
